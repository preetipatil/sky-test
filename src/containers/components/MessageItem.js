import React from 'react';

const mainDiv ={
    height: "150px",
}

const avatarDiv ={
    height: "100px",
    float:"left",
}

class MessageItem extends React.Component {

 constructor(){
    super();
    this.state = {
      emailTxt: "",
    };
 }

 onMouseEnter(email) {   ;
    this.setState({emailTxt: email})
 }

 onMouseLeave() {
    this.setState({emailTxt: ''})
 }

  render() {
    const { chatMessage, email, avatar, timestamp }  = this.props;
    
     return (
         <li >
            <div style={mainDiv}>
                <div style={avatarDiv}>
                    { avatar ? <img  src={avatar}  alt="" /> : " No Avatar "}
                </div>
                <div>
                    <p>DATE {new Date(timestamp).toLocaleString()} </p>
                    <p onMouseEnter={() => this.onMouseEnter(email)} onMouseLeave={() => this.onMouseLeave()}>TXT {chatMessage}  </p>
                    <span>{this.state.emailTxt}</span>
                </div>
            </div>
         </li>
     );
  }
}

export default MessageItem;