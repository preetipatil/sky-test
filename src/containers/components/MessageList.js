import React from 'react';
import MessageItem from './MessageItem';
import PropTypes from 'prop-types';
import { findUserDetails, sortByDate } from '../utils';

function NoMessages() {
  return <div>No messages found</div>;
}

class MessageList extends React.Component {
  renderMessageList({ messages, members }) {
    const posts = sortByDate(messages);
    const findUser = findUserDetails.bind(this, members || []);
    return (
      <ul>
        {posts.map(post => {
          let user = findUser(post.userId);
          return (
            <MessageItem
              key={post.id}
              chatMessage={post.message}
              email={user.email}
              avatar={user.avatar}
              timestamp={post.timestamp}
            />
          );
        })}
      </ul>
    );
  }

  render() {
    return (
      <div>
        <h1>LOADED</h1>
        {this.props.messages ? this.renderMessageList(this.props) : <NoMessages />}
      </div>
    );
  }
}

MessageList.propTypes = {
  messages: PropTypes.array,
  members: PropTypes.array,
};
export default MessageList;
