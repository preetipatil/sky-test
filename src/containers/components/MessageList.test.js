import React from 'react';
import { shallow } from 'enzyme';
import MessageList from './MessageList';
import MessageItem from './MessageItem';

describe('MessageList ', () => {
  it('should render Message Item correctly', () => {
    const props = {
      messages: [
        {
          id: 'cd445e6d-e514-424f-ba8f-16ec842002c6',
          userId: 'fe27b760-a915-475c-80bb-7cdf14cc6ef3',
          message: 'Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla.',
          timestamp: '2017-02-09T04:27:38Z',
        },
      ],
      members: [
        {
          id: 'fe27b760-a915-475c-80bb-7cdf14cc6ef3',
          firstName: 'Albert',
          lastName: 'Rose',
          email: 'arosec@bbb.org',
          avatar: 'http://dummyimage.com/100x100.jpg/5fa2dd/ffffff',
          ip: '20.79.231.223',
        },
      ],
    };
    const tree = shallow(<MessageList {...props} />);
    expect(tree).toMatchSnapshot();
    expect(tree.find('ul').length).toBe(1);
    expect(tree.containsAnyMatchingElements([<MessageItem />])).toEqual(true);
  });
});
