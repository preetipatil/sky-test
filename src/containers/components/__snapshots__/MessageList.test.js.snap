// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList  should render Message Item correctly 1`] = `
<div>
  <h1>
    LOADED
  </h1>
  <ul>
    <MessageItem
      avatar="http://dummyimage.com/100x100.jpg/5fa2dd/ffffff"
      chatMessage="Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla."
      email="arosec@bbb.org"
      key="cd445e6d-e514-424f-ba8f-16ec842002c6"
      timestamp="2017-02-09T04:27:38Z"
    />
  </ul>
</div>
`;
